kfs-tax/src/main/java/edu/arizona/kfs/module/tax/batch/PayeeMasterExtractStep.java

```diff
@@ -0,0 +1,106 @@
+package edu.arizona.kfs.module.tax.batch;
+
+import java.sql.Timestamp;
+import java.util.Collection;
+import java.util.Date;
+import java.util.Set;
+
+import org.kuali.kfs.sys.batch.AbstractStep;
+import org.kuali.kfs.vnd.businessobject.VendorDetail;
+import org.kuali.rice.krad.service.BusinessObjectService;
+
+import edu.arizona.kfs.module.tax.businessobject.ExtractHistory;
+import edu.arizona.kfs.module.tax.service.TaxBatchExtractService;
+import edu.arizona.kfs.module.tax.service.TaxParameterHelperService;
+
+public class PayeeMasterExtractStep extends AbstractStep {
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(PayeeMasterExtractStep.class);
+
+    private TaxBatchExtractService taxBatchExtractService;
+    private TaxParameterHelperService taxParameterHelperService;
+    private BusinessObjectService businessObjectService;
+
+    public void setTaxBatchExtractService(TaxBatchExtractService taxBatchExtractService) {
+        this.taxBatchExtractService = taxBatchExtractService;
+    }
+
+    public void setTaxParameterHelperService(TaxParameterHelperService taxParameterHelperService) {
+        this.taxParameterHelperService = taxParameterHelperService;
+    }
+
+    public void setBusinessObjectService(BusinessObjectService businessObjectService) {
+        this.businessObjectService = businessObjectService;
+    }
+
+    @Override
+    public boolean execute(String jobName, Date jobRunDate) throws InterruptedException {
+        LOG.info("started master payee extraction step of job " + jobName + "at " + new Date());
+        boolean retval = extractPayees();
+        return retval;
+    }
+
+    private boolean extractPayees() {
```

Comment:
You obviously misunderstood my request. The method containing business logic that I wanted moved to a service is extractPayees. The execute method was appropriate as originally coded. The only change I would have expected in the execute method was how the extractPayees method was called:
`boolean retval = taxBatchExtractService.extractPayees();`